[BUG] Wrong rosdep key for SQLite3 - buildfarm fails to find SQLite3

# Bug report

### Steps to reproduce
1. Merge source entry to rosdistro (PR ros/rosdistro#49723)
2. Buildfarm triggers `Jdev__ros2_medkit__ubuntu_noble_amd64` job
3. Build fails on `ros2_medkit_fault_manager`

### Expected behavior

Build succeeds - SQLite3 headers and library are installed by rosdep.

### Actual behavior

```
CMake Error at /usr/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find SQLite3 (missing: SQLite3_INCLUDE_DIR SQLite3_LIBRARY)
```

Full build log: https://build.ros2.org/job/Jdev__ros2_medkit__ubuntu_noble_amd64/1/display/redirect

### Root cause

`ros2_medkit_fault_manager/package.xml` declares `<depend>sqlite3</depend>` which maps to the `sqlite3` CLI tool, not the development library. The correct rosdep key is `libsqlite3-dev` which maps to headers + shared library.

```yaml
# rosdep/base.yaml
sqlite3:        -> debian: [sqlite3]          # CLI tool only
libsqlite3-dev: -> debian: [libsqlite3-dev]   # headers + library
```

### Fix

Change in `src/ros2_medkit_fault_manager/package.xml`:
```xml
<!-- before -->
<depend>sqlite3</depend>

<!-- after -->
<depend>libsqlite3-dev</depend>
```

### Environment
- ros2_medkit version: 0.3.0 (main branch)
- ROS 2 distro: Jazzy Jalisco
- OS: Ubuntu Noble 24.04 (buildfarm)

--- a/src/ros2_medkit_fault_manager/package.xml
+++ b/src/ros2_medkit_fault_manager/package.xml
@@ -13,7 +13,7 @@
   <depend>rclcpp</depend>
   <depend>ros2_medkit_msgs</depend>
   <depend>ros2_medkit_serialization</depend>
-  <depend>sqlite3</depend>
+  <depend>libsqlite3-dev</depend>
   <depend>nlohmann-json-dev</depend>
   <depend>rosbag2_cpp</depend>
   <depend>rosbag2_storage</depend>

